biomage/experiment/download.py

```diff
@@ -0,0 +1,306 @@
+import json
+import os
+from enum import Enum
+from fileinput import filename
+from pathlib import Path
+
+import boto3
+import click
+from botocore.exceptions import ClientError
+
+from ..rds.run import run_rds_command
+from ..utils.constants import (
+    CELLSETS_BUCKET,
+    PROCESSED_FILES_BUCKET,
+    RAW_FILES_BUCKET,
+    SAMPLES_BUCKET,
+)
+from .utils import set_modified_date
+
+output_path = "."
```

Comment:
you have this `output_path` as a global variable and as parameter name in a couple of these functions, which can lead to confusion and bugs. Can you name them differently? Maybe the global variable should be called `default_output_path`? Or maybe there is no need for global variable at all?

Having global variables, where we read and/or change their value in various functions is often not a good idea, because we add more functions in, it becomes hard to trace the variable's value at a given point. This leads to more error-prone and less readable code. You can read more on the internet for more detailed explanation. Here is an example explanation why: https://www.quora.com/Are-global-variables-bad-in-Python

Reply:
deleted